Source/i_sound.c

```diff
@@ -186,33 +187,77 @@ static boolean addsfx(sfxinfo_t *sfx, int channel, int pitch)
    if(sfx->data == NULL || pitch != NORM_PITCH)
    {   
       byte *data;
-      Uint32 samplerate, samplelen;
+      Uint32 samplerate, samplelen, samplecount;
 
       // haleyjd: this should always be called (if lump is already loaded,
       // W_CacheLumpNum handles that for us).
       data = (byte *)W_CacheLumpNum(lump, PU_STATIC);
 
+      // [crispy] Check if this is a valid RIFF wav file
+      if (lumplen > 44 && memcmp(data, "RIFF", 4) == 0 && memcmp(data + 8, "WAVEfmt ", 8) == 0)
+      {
+         // Valid RIFF wav file
+         int check;
+
+         // Make sure this is a PCM format file
+         // "fmt " chunk size must == 16
+         check = data[16] | (data[17] << 8) | (data[18] << 16) | (data[19] << 24);
+         if (check != 16)
+            return false;
+
+         // Format must == 1 (PCM)
+         check = data[20] | (data[21] << 8);
+         if (check != 1)
+            return false;
+
+         // FIXME: can't handle stereo wavs
+         // Number of channels must == 1
+         check = data[22] | (data[23] << 8);
+         if (check != 1)
+            return false;
+
+         samplerate = data[24] | (data[25] << 8) | (data[26] << 16) | (data[27] << 24);
+         samplelen = data[40] | (data[41] << 8) | (data[42] << 16) | (data[43] << 24);
+
+         if (samplelen > lumplen - 44)
+            samplelen = lumplen - 44;
+
+         bits = data[34] | (data[35] << 8);
+
+         // Reject non 8 or 16 bit
+         if (bits != 16 && bits != 8)
+            return false;
+
+         data += 44 - 8;
```

Comment:
You increase the `data` pointer here, that's why it fails to free it later on.

Reply:
Fixed in https://github.com/fabiangreffrath/woof/pull/122/commits/eb7fbf9eaea1ca6a8e213ea9a5d44d33abaa9567